libsql-server/src/auth/user_auth_strategies/mod.rs

```diff
@@ -2,15 +2,68 @@ pub mod disabled;
 pub mod http_basic;
 pub mod jwt;
 
-use axum::http::HeaderValue;
-pub use disabled::*;
-pub use http_basic::*;
-pub use jwt::*;
+pub use disabled::Disabled;
+pub use http_basic::HttpBasic;
+pub use jwt::Jwt;
 
 use super::{AuthError, Authenticated};
 
+#[derive(Debug)]
 pub struct UserAuthContext {
-    pub user_credential: Option<HeaderValue>,
+    scheme: Option<String>,
+    token: Option<String>,
+}
+
+impl UserAuthContext {
+    pub fn scheme(&self) -> &Option<String> {
+        &self.scheme
+    }
+
+    pub fn token(&self) -> &Option<String> {
+        &self.token
+    }
+
+    pub fn empty() -> UserAuthContext {
+        UserAuthContext {
+            scheme: None,
+            token: None,
+        }
+    }
+
+    pub fn basic(creds: &str) -> UserAuthContext {
+        UserAuthContext {
+            scheme: Some("Basic".into()),
+            token: Some(creds.into()),
+        }
+    }
+
+    pub fn bearer(token: &str) -> UserAuthContext {
+        UserAuthContext {
+            scheme: Some("Bearer".into()),
+            token: Some(token.into()),
+        }
+    }
+
+    pub fn bearer_opt(token: Option<String>) -> UserAuthContext {
+        UserAuthContext {
+            scheme: Some("Bearer".into()),
+            token: token,
+        }
+    }
+
+    pub fn new(scheme: &str, token: &str) -> UserAuthContext {
+        UserAuthContext {
+            scheme: Some(scheme.into()),
+            token: Some(token.into()),
+        }
+    }
+
+    pub fn from_auth_str(auth_string: &str) -> Result<Self, AuthError> {
+        let (scheme, token) = auth_string
+            .split_once(' ')
```

Comment:
Related to my other comment I see now what the problem is. I think we should support an empty auth string mostly because this is the workflow we have for a ton of tools outside of libsql and in tests. The problem is that now that the parsing of the header is separate from the authentication method we don't know that auth is disabled, because we should throw this error if jwt is enabled and we can't split. 

I think the solution here would be to not return an error here but somehow represent the state within the auth context of an empty auth header and let the auth strategy decide what to do with that. This would allow us to revert all the `dummy_token` stuff and fix the encryption test, as well as ensure that all our other tools continue to work locally.

Reply:
Related to https://github.com/tursodatabase/libsql/pull/1130#discussion_r1513163735

Reply:
full agreement - in general terms, we should not require auth context for strategies that don't use it
I figured we could achieve it by lazy access - passing a function that calculates the context into authenticate
it took me down a massive rabbit hole of dynamic dispatch/mutability/ownership issues
decided it is not worth it, we can (and we will) revisit in later PRs
I propose we turn `authenticate(&self, context: UserAuthContext)` into `authenticate(&self, context: Result<UserAuthContext>)`
auth strategies that need the context will unwrap it and map errors accordingly
auth strategies that don't need it will simply ignore
bringing it up as it is not a common pattern in rust to accept Result as input 
it does fit the bill in this case very well
again, this is just a step in evolving auth layer towards more cohesive style

Reply:
Yeah, I think this makes a lot of sense. Lets do it.